crypto/s2n_certificate.h

```diff
@@ -53,7 +53,7 @@ typedef enum {
     S2N_CERT_TYPE_ECDSA_FIXED_ECDH = 66,
 } s2n_cert_type;
 
-struct s2n_cert_public_key {
```

Comment:
Hah, yes. You both are completely right. I've been wanted to address #538 anyways, and this makes a lot more sense.